packages/flutter/lib/src/cupertino/tab_scaffold.dart

```diff
@@ -121,14 +128,30 @@ class CupertinoTabScaffold extends StatefulWidget {
   /// In that case, the child's [BuildContext]'s [MediaQuery] will have a
   /// bottom padding indicating the area of obstructing overlap from the
   /// [tabBar].
+  ///
+  /// Must not be null.
   final IndexedWidgetBuilder tabBuilder;
 
   @override
   _CupertinoTabScaffoldState createState() => new _CupertinoTabScaffoldState();
 }
 
 class _CupertinoTabScaffoldState extends State<CupertinoTabScaffold> {
-  int _currentPage = 0;
+  int _currentPage;
+
+  @override
+  void initState() {
+    super.initState();
+    _currentPage = widget.tabBar.currentIndex;
+  }
+
+  @override
+  void didUpdateWidget(CupertinoTabScaffold oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (widget.tabBar.currentIndex != oldWidget.tabBar.currentIndex) {
+      setState(() => _currentPage = widget.tabBar.currentIndex);
```

Comment:
You don't need to call `setState` here.

Reply:
ya, totally. Thanks!

(we should add a debug time assert or a linter :))